### PR TITLE
fix HTTPInstaller not handling octet-stream media types

### LINKED ISSUE #11272: How to create and install plugin tgz
Is the plugin installation with tgz still supported/working? It's mentioned in documentation at https://helm.sh/docs/topics/plugins/#installing-a-plugin

But when trying to download a .tgz from github releases this happens:

```
$ helm plugin install https://github.com/xxxx/xxxx/releases/download/xx-xx-xx/helm-plugin.tgz
Error: Unable to get repository: Cloning into '/Users/mblaschke/Library/Caches/helm/plugins/https-github.com-xxxx-helm-xxxxx-releases-download-xx-xx-xx-helm-plugin.tgz'...
remote: Not Found
fatal: repository 'https://github.com/xxxx/xxxx/releases/download/xx-xx-xx/helm-plugin.tgz/' not found
: exit status 128
```

when fetching the tgz and trying it locally it also fails:

```
$ helm --debug plugin install release-assets/helm-plugin.tgz
Error: plugin metadata (plugin.yaml) missing
helm.go:84: [debug] plugin metadata (plugin.yaml) missing
helm.sh/helm/v3/pkg/plugin/installer.init
	helm.sh/helm/v3/pkg/plugin/installer/installer.go:32
runtime.doInit
	runtime/proc.go:6222
runtime.doInit
	runtime/proc.go:6199
runtime.main
	runtime/proc.go:233
runtime.goexit
	runtime/asm_arm64.s:1263


$ tar tzvf release-assets/helm-plugin.tgz | grep plugin
-rw-r--r--  0 mblaschke staff      693 Aug 25 15:39 plugin.yaml


$ file release-assets/helm-plugin.tgz
release-assets/helm-plugin.tgz: gzip compressed data, last modified: Thu Aug 25 13:39:09 2022, from Unix, original size modulo 2^32 73625600
```

might be also related to #10402

Output of `helm version`:
version.BuildInfo{Version:"v3.9.2", GitCommit:"1addefbfe665c350f4daf868a9adc5600cc064fd", GitTreeState:"clean", GoVersion:"go1.18.4"

Output of `kubectl version`:
n/A

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
n/A

### PR DESCRIPTION
**Pull request by s7m4b4**
partial-fix #11272
Signed-off-by: Stephen Bell <8148044+s7m4b4@users.noreply.github.com>

**What this PR does / why we need it**:
As mentioned in #11272, Github uses "application/octet-stream" media type which `mediaTypeToExtension` does not consider, meaning any plugin installs from Github will fail when using `HTTPInstaller`. This fixes that.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by hickeyma — APPROVED** [review]
@s7m4b4 Thank you for pushing this fix.

I have changed the description to say partial-fix #11272 as this fix fixes the HTTP get issue. As you mentioned in https://github.com/helm/helm/issues/11272#issuecomment-1258822058, there are more open issues when files are not in the root of the archive.

**Comment by hickeyma** [comment]
Would like another maintainer to take a look also before merging.

**Comment by hickeyma** [comment]
Thanks for the feedback @phil9909 and input always welcome. Lets get more input and see how we want to handle this.

File: pkg/plugin/installer/http_installer.go
Comment (mattfarina) [inline pkg/plugin/installer/http_installer.go:67]: The addition of application/octet-stream doesn't mean it's a valid file. We need to make sure the extension is valid in the case of this. Can we pass it in to this function and then check for this case that the extension is right?

**Review by mattfarina — CHANGES_REQUESTED** [review]
I tried this change on top of https://github.com/helm/helm/pull/11508 and it doesn't fix the issue. It also opens the door to downloading other things and trying to work on them. We shouldn't do that.

**Comment by joejulian** [comment]
Closing this as abandoned. If you can come back and address the change request, please reopen the PR.

**Pull request closed without merging**
